error: this `MutexGuard` is held across an `await` point
  --> $DIR/await_holding_lock.rs:8:13
   |
LL |         let guard = x.lock().unwrap();
   |             ^^^^^
   |
   = note: `-D clippy::await-holding-lock` implied by `-D warnings`
   = help: consider using an async-aware `Mutex` type or ensuring the `MutexGuard` is dropped before calling await
note: these are all the `await` points this lock is held through
  --> $DIR/await_holding_lock.rs:8:9
   |
LL | /         let guard = x.lock().unwrap();
LL | |         baz().await
LL | |     }
   | |_____^

error: this `MutexGuard` is held across an `await` point
  --> $DIR/await_holding_lock.rs:23:13
   |
LL |         let guard = x.read().unwrap();
   |             ^^^^^
   |
   = help: consider using an async-aware `Mutex` type or ensuring the `MutexGuard` is dropped before calling await
note: these are all the `await` points this lock is held through
  --> $DIR/await_holding_lock.rs:23:9
   |
LL | /         let guard = x.read().unwrap();
LL | |         baz().await
LL | |     }
   | |_____^

error: this `MutexGuard` is held across an `await` point
  --> $DIR/await_holding_lock.rs:28:13
   |
LL |         let mut guard = x.write().unwrap();
   |             ^^^^^^^^^
   |
   = help: consider using an async-aware `Mutex` type or ensuring the `MutexGuard` is dropped before calling await
note: these are all the `await` points this lock is held through
  --> $DIR/await_holding_lock.rs:28:9
   |
LL | /         let mut guard = x.write().unwrap();
LL | |         baz().await
LL | |     }
   | |_____^

error: this `MutexGuard` is held across an `await` point
  --> $DIR/await_holding_lock.rs:53:13
   |
LL |         let guard = x.lock().unwrap();
   |             ^^^^^
   |
   = help: consider using an async-aware `Mutex` type or ensuring the `MutexGuard` is dropped before calling await
note: these are all the `await` points this lock is held through
  --> $DIR/await_holding_lock.rs:53:9
   |
LL | /         let guard = x.lock().unwrap();
LL | |
LL | |         let second = baz().await;
LL | |
...  |
LL | |         first + second + third
LL | |     }
   | |_____^

error: this `MutexGuard` is held across an `await` point
  --> $DIR/await_holding_lock.rs:66:17
   |
LL |             let guard = x.lock().unwrap();
   |                 ^^^^^
   |
   = help: consider using an async-aware `Mutex` type or ensuring the `MutexGuard` is dropped before calling await
note: these are all the `await` points this lock is held through
  --> $DIR/await_holding_lock.rs:66:13
   |
LL | /             let guard = x.lock().unwrap();
LL | |             baz().await
LL | |         };
   | |_________^

error: this `MutexGuard` is held across an `await` point
  --> $DIR/await_holding_lock.rs:78:17
   |
LL |             let guard = x.lock().unwrap();
   |                 ^^^^^
   |
   = help: consider using an async-aware `Mutex` type or ensuring the `MutexGuard` is dropped before calling await
note: these are all the `await` points this lock is held through
  --> $DIR/await_holding_lock.rs:78:13
   |
LL | /             let guard = x.lock().unwrap();
LL | |             baz().await
LL | |         }
   | |_________^

error: this `MutexGuard` is held across an `await` point
  --> $DIR/await_holding_lock.rs:89:13
   |
LL |         let guard = x.lock();
   |             ^^^^^
   |
   = help: consider using an async-aware `Mutex` type or ensuring the `MutexGuard` is dropped before calling await
note: these are all the `await` points this lock is held through
  --> $DIR/await_holding_lock.rs:89:9
   |
LL | /         let guard = x.lock();
LL | |         baz().await
LL | |     }
   | |_____^

error: this `MutexGuard` is held across an `await` point
  --> $DIR/await_holding_lock.rs:104:13
   |
LL |         let guard = x.read();
   |             ^^^^^
   |
   = help: consider using an async-aware `Mutex` type or ensuring the `MutexGuard` is dropped before calling await
note: these are all the `await` points this lock is held through
  --> $DIR/await_holding_lock.rs:104:9
   |
LL | /         let guard = x.read();
LL | |         baz().await
LL | |     }
   | |_____^

error: this `MutexGuard` is held across an `await` point
  --> $DIR/await_holding_lock.rs:109:13
   |
LL |         let mut guard = x.write();
   |             ^^^^^^^^^
   |
   = help: consider using an async-aware `Mutex` type or ensuring the `MutexGuard` is dropped before calling await
note: these are all the `await` points this lock is held through
  --> $DIR/await_holding_lock.rs:109:9
   |
LL | /         let mut guard = x.write();
LL | |         baz().await
LL | |     }
   | |_____^

error: this `MutexGuard` is held across an `await` point
  --> $DIR/await_holding_lock.rs:134:13
   |
LL |         let guard = x.lock();
   |             ^^^^^
   |
   = help: consider using an async-aware `Mutex` type or ensuring the `MutexGuard` is dropped before calling await
note: these are all the `await` points this lock is held through
  --> $DIR/await_holding_lock.rs:134:9
   |
LL | /         let guard = x.lock();
LL | |
LL | |         let second = baz().await;
LL | |
...  |
LL | |         first + second + third
LL | |     }
   | |_____^

error: this `MutexGuard` is held across an `await` point
  --> $DIR/await_holding_lock.rs:147:17
   |
LL |             let guard = x.lock();
   |                 ^^^^^
   |
   = help: consider using an async-aware `Mutex` type or ensuring the `MutexGuard` is dropped before calling await
note: these are all the `await` points this lock is held through
  --> $DIR/await_holding_lock.rs:147:13
   |
LL | /             let guard = x.lock();
LL | |             baz().await
LL | |         };
   | |_________^

error: this `MutexGuard` is held across an `await` point
  --> $DIR/await_holding_lock.rs:159:17
   |
LL |             let guard = x.lock();
   |                 ^^^^^
   |
   = help: consider using an async-aware `Mutex` type or ensuring the `MutexGuard` is dropped before calling await
note: these are all the `await` points this lock is held through
  --> $DIR/await_holding_lock.rs:159:13
   |
LL | /             let guard = x.lock();
LL | |             baz().await
LL | |         }
   | |_________^

error: aborting due to 12 previous errors

